### PR TITLE
[allplay] Make sure AllPlay binding reads configuration correctly

### PR DESCRIPTION
**Pull request by dominicdesu**
Fixes issue #2335 that the AllPlay binding did not read the configuration correctly. Also fixes reading of the double values specified in the configuration (did not work correctly after the first issue has been fixed)

### CODE REVIEW

**Comment by martinvw** [comment]
Refrigerator build

File: addons/binding/org.openhab.binding.allplay/src/main/java/org/openhab/binding/allplay/internal/AllPlayBindingProperties.java [hunk 1]
```diff
@@ -17,25 +17,34 @@
...
+
+        rewindSkipTimeInSec = getDoubleProperty(properties, REWIND_SKIP_TIME_PROPERTY, REWIND_SKIP_TIME_DEFAULT_VALUE);
+        fastForwardSkipTimeInSec = getDoubleProperty(properties, FAST_FORWARD_SKIP_TIME_PROPERTY,
+                FAST_FORWARD_SKIP_TIME_DEFAULT_VALUE);
         callbackUrl = (String) properties.get(CALLBACK_URL);
     }
 
-    public int getRewindSkipTimeInSec() {
+    private double getDoubleProperty(Dictionary<String, Object> properties, String propertyKey, double defaultValue) {
+        String value = (String) properties.get(propertyKey);
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.allplay/src/main/java/org/openhab/binding/allplay/internal/AllPlayBindingProperties.java]: Careful - the configuration can potentially contain double already (should be the case if configured through the Paper UI), it is not guaranteed that you will find only Strings (which is the case if configured through a cfg file).

File: addons/binding/org.openhab.binding.allplay/src/main/java/org/openhab/binding/allplay/internal/AllPlayBindingProperties.java [hunk 2]
```diff
@@ -17,25 +17,34 @@
  */
 public class AllPlayBindingProperties {
 
-    private final Integer rewindSkipTimeInSec;
-    private final Integer fastForwardSkipTimeInSec;
+    private final double rewindSkipTimeInSec;
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.allplay/src/main/java/org/openhab/binding/allplay/internal/AllPlayBindingProperties.java]: Do you really need doubles here? Or could you simply change the parameter declaration to be Integer (in the binding.xml)?

Comment (dominicdesu) [inline addons/binding/org.openhab.binding.allplay/src/main/java/org/openhab/binding/allplay/internal/AllPlayBindingProperties.java]: Hi, thanks for the feedback!
I just thought double would leave more flexibility to the user. I can easily change it to Integer. Out of curiosity: What is the problem with double here?

Comment (dominicdesu) [inline addons/binding/org.openhab.binding.allplay/src/main/java/org/openhab/binding/allplay/internal/AllPlayBindingProperties.java]: Even after changing the parameter type to integer in the binding.xml, PaperUI still allows me to insert decimal numbers such as 10.5. Do I need to consider this too?

Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.allplay/src/main/java/org/openhab/binding/allplay/internal/AllPlayBindingProperties.java]: I would have hoped that this is already fixed by https://github.com/eclipse/smarthome/pull/4001 - can you check if you have the lastet ESH stable in your target platform? (should be from 20170815)

Comment (dominicdesu) [inline addons/binding/org.openhab.binding.allplay/src/main/java/org/openhab/binding/allplay/internal/AllPlayBindingProperties.java]: Indeed I didn't have the latest version of ESH. Unfortunately, upgrading didn't help. applyParameterType is only called when I open the config dialog in the PaperUI, not when I save the new values. The other two adjusted functions in eclipse/smarthome#4001 are only calling parseInt on the default value, not on the user-entered value. Seems like the fix is only for default values?

Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.allplay/src/main/java/org/openhab/binding/allplay/internal/AllPlayBindingProperties.java]: @htreu Could you check whether there is still some bug regarding integer config parameter input fields in the Paper UI?

Comment (htreu) [inline addons/binding/org.openhab.binding.allplay/src/main/java/org/openhab/binding/allplay/internal/AllPlayBindingProperties.java]: I will look into it. The fix in https://github.com/eclipse/smarthome/pull/4001 was indeed about defaults and option values.

Comment (dominicdesu) [inline addons/binding/org.openhab.binding.allplay/src/main/java/org/openhab/binding/allplay/internal/AllPlayBindingProperties.java]: @kaikreuzer, As I need to handle invalid string values anyway (if configured through cfg file), I have changed the implementation and created a new PR.
Btw. the property dictionary always gives me string values, even if set through PaperUI (in opposition to your other review comment). I wasn't able to get Integers in the config. Anyway, implementation should be on the safe side now.

Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.allplay/src/main/java/org/openhab/binding/allplay/internal/AllPlayBindingProperties.java]: sounds fair, thanks.

Comment (htreu) [inline addons/binding/org.openhab.binding.allplay/src/main/java/org/openhab/binding/allplay/internal/AllPlayBindingProperties.java]: for reference: integer config parameters should not allow decimal input: https://github.com/eclipse/smarthome/issues/4115

Comment (htreu) [inline addons/binding/org.openhab.binding.allplay/src/main/java/org/openhab/binding/allplay/internal/AllPlayBindingProperties.java]: This bug can be worked around when giving a pattern of `\d+` for the parameter like `<parameter name="myInt" type="integer" pattern="\d+">`. PaperUI will respect the pattern and validate the input.

**Pull request merged**